starfish/core/image/Filter/ilastik_pre_trained_probability.py

```diff
@@ -0,0 +1,114 @@
+import subprocess
+import tempfile
+from pathlib import Path
+from typing import Optional, Union
+
+import h5py
+import numpy as np
+import scipy.ndimage as ndi
+from skimage.filters import threshold_otsu
+
+from starfish.core.imagestack.imagestack import ImageStack
+from ._base import FilterAlgorithm
+
+
+class IlastikPretrainedProbability(FilterAlgorithm):
+    """
+    Use an existing ilastik pixel classifier to generate a probability image for a dapi image.
+    NOTE: This api may not be used without a downloaded and installed version of Ilastik. Visit
+    https://www.ilastik.org/download.html to download.
+
+    Parameters
+    ----------
+    ilastik_executable: Union[Path, str]
+        Path to run_ilastik.sh needed for running ilastik in headless mode. Typically the script is
+        located: "/Applications/{ILASTIK_VERSION}/Contents/ilastik-release/run_ilastik.sh"
+
+    ilastik_project: Union[Path, str]
+        path to ilastik project .ilp file
+
+    """
+
+    def __init__(self, ilastik_executable: Union[Path, str], ilastik_project: Union[Path, str]):
+        ilastik_executable = ilastik_executable \
+            if isinstance(ilastik_executable, Path) else Path(ilastik_executable)
+        if not ilastik_executable.exists():
+            raise EnvironmentError("Can not find run_ilastik.sh. Make sure you've provided the "
+                                   "correct location. If you need to download ilastik please"
+                                   " visit: https://www.ilastik.org/download.html")
+        self.ilastik_executable = ilastik_executable
+        self.ilastik_project = ilastik_project
+
+    def run(
+            self,
+            stack: ImageStack,
+            in_place: bool = False,
+            verbose: bool = False,
+            n_processes: Optional[int] = None,
+            *args,
+    ) -> Optional[ImageStack]:
+        """
+        Use a pre trained probability pixel classification model to generate probabilites
+        for a dapi image
+
+        Parameters
+        ----------
+        stack : ImageStack
+            Dapi image to be run through ilastik.
+        in_place : bool
+            N/A
+        verbose : bool
+            N/A
+        n_processes : Optional[int]
+            N/A
+
+        Returns
+        -------
+        ImageStack :
+            A new ImageStack created from the cell probabilities provided by ilastik.
+        """
+
+        # temp files
+        temp_dir = tempfile.TemporaryDirectory()
+        dapi_file = f"{temp_dir.name}_dapi.npy"
+        output_file = f"{temp_dir.name}_dapi_Probabilities.h5"
+        np.save(dapi_file, stack.xarray.values.squeeze())
+
+        # env {} is needed to fix the weird virtualenv stuff
+        subprocess.run(
```

Comment:
do you want `subprocess.check_call`?  or you can pass `check=True` to `subprocess.run`